### PR TITLE
Update skyplot alex

### LINKED ISSUE #247: Add animations to skyplot
As an amateur astronomer I like moving images of space, when I'm stargazing I want to be able to see the sky as it changes in a skyplot so that I can plan my stargazing accordingly

- Update the skyplot function to be animated

### PR DESCRIPTION
**Pull request by AlexOBarnes**
This closes #247 #248 

What I have done:
- Updated skyplot to be animated
- Updated the skyplot query to query data from over one night

### CODE REVIEW

File: dashboard/test_image_pipeline.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def test_get_iss_location_success(self, mock_get):
             'iss_position': {'latitude': '46.9566','longitude': '-3.7535'}}
         mock_get.return_value = mock_response
         result = get_iss_location()
-        expected_result = {'timestamp': dt(2024,10,14,12,26,12),
+        expected_result = {'timestamp': dt(2024,10,14,13,26,12),
```
Comment (AlexOBarnes) [inline dashboard/test_image_pipeline.py:154]:
```suggestion
        expected_result = {'timestamp': dt(2024,10,14,12,26,12),
```

File: dashboard/azimuth_plot.py [hunk 2]
```diff
@@ -12,77 +13,137 @@
...
 
-def sky_plot(celestial_df: pd.DataFrame, region: str, time_point: dt) -> plt.subplot:
-    '''Plots the azimuth and altitude data on a radial plot'''
+def sky_plot(celestial_df: pd.DataFrame, region: str, time_point) -> plt.subplot:
+    '''Constructs the sky_plot using the celestial data'''
     fig, ax = plt.subplots(figsize=(8, 8), subplot_kw={
                            'projection': 'polar'}, facecolor='black')
-
     ax.set_facecolor('black')
+    body_properties_map = {'Sun': {'color': 'gold', 'size': 1391000},
```
Comment (MurasatoNaoya) [inline dashboard/azimuth_plot.py:47]: Each body now has an associated colour, nice aesthetic change.

File: dashboard/azimuth_plot.py [hunk 3]
```diff
@@ -12,77 +13,137 @@
...
+    for i in range(len(current_df)):
+        text = plt.text(current_df['azimuth'].iloc[i],
+                        np.radians(current_df['altitude']).iloc[i],
+                        current_df['body_name'].iloc[i],
+                        fontsize=10, ha='right', va='bottom', color='white')
+        text_elements.append(text)
+    ax.set_title(f'Skyplot for {region} at {time_point.strftime("%d/%m/%Y %H:%M")}', color='white')
+
+
+def animate_skyplot(celestial_df: pd.DataFrame, region: str):
```
Comment (MurasatoNaoya) [inline dashboard/azimuth_plot.py:128]: Interesting use of an external library to animate the visualisation, very cool.

**Review by MurasatoNaoya — APPROVED** [review]
LGTM.

File: dashboard/azimuth_plot.py (hunk 2 above)
Comment (MurasatoNaoya) [inline dashboard/azimuth_plot.py:47]: * colour

File: dashboard/azimuth_plot.py [hunk 4]
```diff
@@ -12,77 +13,137 @@
...
+    data = get_azimuth_data(region, time)
     clean_data = transform_azimuth_data(data)
-    sky_plot(clean_data,region,time)
+    animate_skyplot(clean_data, region)
 
 
 if __name__ == '__main__':
     load_dotenv()
-    make_sky_plot('Scotland',dt(2024,10,26,20,0,0))
+    make_sky_plot('Scotland', dt(2024, 10, 25, 0, 0, 0))
```
Comment (MurasatoNaoya) [inline dashboard/azimuth_plot.py:149]: Please implement all of this in assembly, thanks.

Comment (AlexOBarnes) [reply dashboard/azimuth_plot.py:149]: Of course, I'll make a ticket

**Review by CalebMathers — APPROVED** [review]

**Pull request merged**